### PR TITLE
Martijn mapping fixed

### PR DESCRIPTION
**Pull request by ThomasvLingen**
kill me

### CODE REVIEW

File: src/gamelogic/ControlMapper.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ControlMapper::~ControlMapper()
+    {
+    }
+
+    void ControlMapper::handle_input(Input input)
+    {
+        this->_input_actions.actions_off.clear();
+        this->_input_actions.actions_on.clear();
+        for (auto const& action_key_pair : this->_mapping_info)
+        {
```
Comment (ThomasvLingen) [inline src/gamelogic/ControlMapper.cpp]: should not be on new line

File: src/gamelogic/ControlMapper.cpp [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            if (std::find(input.keys_down.begin(), input.keys_down.end(), action_key_pair.second) != input.keys_down.end()) {
+                //action key found in keys_down
+                this->_input_actions.actions_on.push_back(action_key_pair.first);
+                //input_actions.actions_on.push_back(action_key_pair.first);
```
Comment (ThomasvLingen) [inline src/gamelogic/ControlMapper.cpp]: Remove this comment

File: src/gamelogic/ControlMapper.cpp [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            }
+
+            if (std::find(input.keys_released.begin(), input.keys_released.end(), action_key_pair.second) != input.keys_released.end()) {
+                //action key found in keys_released
+                //input_actions.actions_off.push_back(action_key_pair.first);
```
Comment (ThomasvLingen) [inline src/gamelogic/ControlMapper.cpp]: Remove this comment

File: src/gamelogic/Player.hpp [hunk 4]
```diff
@@ -39,6 +40,8 @@ namespace GameLogic {
     private:
         SPTR_Level _level;
 
+        ControlMapper _control_mapper;
```
Comment (ThomasvLingen) [inline src/gamelogic/Player.hpp]: This should be a ref, right?

File: src/gamelogic/Player.cpp [hunk 5]
```diff
@@ -6,9 +6,10 @@
 
 namespace GameLogic {
 
-    Player::Player(CoordinateDouble position)
+    Player::Player(CoordinateDouble position, ControlMapper control_mapper)
```
Comment (ThomasvLingen) [inline src/gamelogic/Player.cpp]: This should also be a ref

File: src/gamelogic/state/HelpState.cpp [hunk 6]
```diff
@@ -36,23 +40,15 @@ namespace State {
         this->_context.SDL_facade.draw_image("res/helpscreen.bmp" , {0,0});
         this->_context.SDL_facade.draw_text("Controls for Vidya Game", FontType::alterebro_pixel, this->_color, {120, 50});
 
+        ControlMapper mapper = this->_context.control_mapper;
```
Comment (ThomasvLingen) [inline src/gamelogic/state/HelpState.cpp]: This should be a ref, right? Otherwise it's copied

File: src/gamelogic/state/MappingState.cpp [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+        this->_context.SDL_facade.draw_rect({0,0}, this->_context.SDL_facade.get_width(), this->_context.SDL_facade.get_height(),{0,0,0});
+
+        this->_context.SDL_facade.draw_text("Mapping of controls", FontType::alterebro_pixel_plus, this->_color, {50, 15});
+
+        this->_context.SDL_facade.handle_sdl_events();
+
+        Input keys = this->_context.SDL_facade.get_input();
+
+        if (_action_is_selected) {
```
Comment (ThomasvLingen) [inline src/gamelogic/state/MappingState.cpp]: this->

File: src/gamelogic/ControlMapper.cpp (hunk 1 above)
Comment (WoutervanGeel) [inline src/gamelogic/ControlMapper.cpp]: Done

File: src/gamelogic/ControlMapper.cpp (hunk 3 above)
Comment (WoutervanGeel) [inline src/gamelogic/ControlMapper.cpp]: Done

File: src/gamelogic/state/MappingState.cpp (hunk 7 above)
Comment (WoutervanGeel) [inline src/gamelogic/state/MappingState.cpp]: Done

File: src/gamelogic/Player.hpp (hunk 4 above)
Comment (WoutervanGeel) [inline src/gamelogic/Player.hpp]: Done

File: src/gamelogic/Player.cpp (hunk 5 above)
Comment (WoutervanGeel) [inline src/gamelogic/Player.cpp]: Done

File: src/gamelogic/state/HelpState.cpp (hunk 6 above)
Comment (WoutervanGeel) [inline src/gamelogic/state/HelpState.cpp]: Done

**Comment by ThomasvLingen** [comment]
As discussed with @WoutervanGeel @jorgdebont and me, this requires some refactorings if we don't want to completely throw our Handleable idea out of the bloody window.

Yell at me when that's done, @WoutervanGeel

**Pull request closed without merging**